### PR TITLE
[PR-into-PR] Monitor E2E rewrite - validator

### PR DESCRIPTION
**Pull request by rzadp**
https://github.com/poanetwork/tokenbridge/pull/187#discussion_r313584109

Extending the https://github.com/poanetwork/tokenbridge/pull/187 with a little more explicit testing of the validators

### CODE REVIEW

File: monitor-e2e/test/common.js [hunk 1]
```diff
@@ -43,9 +43,21 @@ types.forEach(type => {
...
+
+      it('foreign', () => {
+        assert(typeof data.foreign.validators === 'object')
+        assert(data.foreign.validators[validator.address].balance > 0)
+        assert(data.foreign.validators[validator.address].leftTx > 0)
+        assert(data.foreign.validators[validator.address].gasPrice > 0)
+      })
+
+      it('requiredSignaturesMatch', () => assert(data.requiredSignaturesMatch, 1))
       it('validatorsMatch', () => assert(data.validatorsMatch))
```
Comment (akolotov) [inline monitor-e2e/test/common.js]: Consider to change the number of validators on one side of the bridge by invoking `addValidator` from the validators contract. The monitor should have `false` in `validatorsMatch` when the transaction is included in the block.

Comment (rzadp) [reply monitor-e2e/test/common.js]: Right, will do.

Comment (rzadp) [reply monitor-e2e/test/common.js]: I have done the changes in similar fashion as #191 

1. The precondition is that `validatorsMatch === true`
2. Then the `addValidator` is executed on only one side of the bridge
3. After `checkAll` and `axios.get`, the expected result is that `validatorsMatch === false`

Comment (akolotov) [reply monitor-e2e/test/common.js]: Thanks!

File: Dockerfile.e2e [hunk 2]
```diff
@@ -1,5 +1,7 @@
 FROM node:10
 
+RUN curl -sSL https://get.docker.com/ | sh
```
Comment (akolotov) [inline Dockerfile.e2e]: Let's wait for final decision in https://github.com/poanetwork/tokenbridge/pull/191. I provided there few options how we can avoid docker-in-docker environment.

**Comment by rzadp** [comment]
@akolotov This has been updated after #191 and is ready again for review.

**Pull request merged**